### DISCUSSION TITLE
On the ARM platform, the following code has issues.

### DISCUSSION THREAD

**Discussion by stdhans** [daniele77/cli#220] [General]
Hi daniele, On the ARM platform, the following code has issues.
in  file: genericasioremotecli.h（I added 'signed char' because -1 != 0xFF in arm32bit):

   void Consume(signed char c)
    {        
        if (escape)
        {
            if (c == (signed char)(IAC))
                Data(c);
            else
                Command(c);
            escape = false;
        }
        else
        {
            if (c == (signed char)(IAC))
                escape = true;
            else
                Data(c);
        }
    }

**Comment by daniele77** [comment]
This should be fixed with commit: 2921758
Please, check if everything is ok and let me know.
Thanks.

**Comment by stdhans** [comment]
The problem has been resolved, thank you for providing such excellent open-source resources.



---- Replied Message ----
| From | Daniele ***@***.***> |
| Date | 02/03/2024 02:26 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [daniele77/cli] On the ARM platform, the following code has issues. (Discussion #220) |

Closed #220 as resolved.
